app/models/concerns/project_media_cached_fields.rb

```diff
@@ -644,7 +644,8 @@ def recalculate_report_status
     end
 
     def recalculate_tags_as_sentence
-      self.get_annotations('tag').map(&:load).map(&:tag_text).uniq.join(', ')
+      tag_text_ids = Tag.where(annotation_type: 'tag', annotated_type: 'ProjectMedia', annotated_id: self.id).collect{|t| t.data['tag']}
+      TagText.where(id: tag_text_ids).map(&:text).join(', ')
```

Comment:
Suggestion: changing `map(&:text)` to `pluck(:text)`, this way we only load the value for text instead, instead of the whole `TagText` object.